src/Optimization/hiopAlgFilterIPM.cpp

```diff
@@ -1485,6 +1529,170 @@ void hiopAlgFilterIPMQuasiNewton::outputIteration(int lsStatus, int lsNum, int u
   }
 }
 
+#ifdef HIOP_USE_AXOM
+
+bool hiopAlgFilterIPMQuasiNewton::save_state_to_file(const ::std::string& path) noexcept
+{
+  auto success = true;
+  sidre::DataStore* ds = nullptr;
+  try {
+    ds = new sidre::DataStore();
+    this->save_state_to_data_store(ds);
+
+    sidre::IOManager writer(this->get_nlp()->get_comm());
+    int n_files;
+    MPI_Comm_size(this->get_nlp()->get_comm(), &n_files);
+    writer.write(ds->getRoot(), n_files, path.c_str(), sidre::Group::getDefaultIOProtocol());
+  } catch(const std::exception& exp) {
+    nlp->log->printf(hovError, "Error in saving checkpoint to file '%s'\n", path.c_str());
+    nlp->log->printf(hovError,  "  Addtl info: %s\n", exp.what());
+    success = false;    
+  }  
+  delete ds;
```

Comment:
```suggestion
  sidre::DataStore ds;
  try {
    this->save_state_to_data_store(ds);

    sidre::IOManager writer(this->get_nlp()->get_comm());
    int n_files;
    MPI_Comm_size(this->get_nlp()->get_comm(), &n_files);
    writer.write(ds->getRoot(), n_files, path.c_str(), sidre::Group::getDefaultIOProtocol());
  } catch(const std::exception& exp) {
    nlp->log->printf(hovError, "Error in saving checkpoint to file '%s'\n", path.c_str());
    nlp->log->printf(hovError,  "  Addtl info: %s\n", exp.what());
    success = false;    
  }  
```
There is no need to use dynamic memory here.